src/mlpack/tests/mock_categorical_data.hpp

```diff
@@ -113,4 +113,62 @@ inline void MockCategoricalData(arma::mat& d,
   }
 }
 
+/**
+ * Create a mock categorical dataset for testing regression.
+ */
+inline void MockCategoricalData(arma::mat& d,
+                                arma::Row<double>& l,
+                                mlpack::data::DatasetInfo& datasetInfo)
+{
+  // Dataset of size 4000.
+  d.set_size(5, 4000);
+  l.set_size(4000);
+
+  for (size_t i = 0; i < 4000; ++i)
+  {
+    // Random numeric features.
+    d(0, i) = mlpack::math::Random();
+    d(1, i) = mlpack::math::Random(-1, 1);
+    d(2, i) = mlpack::math::Random();
+
+    // Binary feature.
+    d(3, i) = mlpack::math::RandInt(0, 2);
+    // 5-category categorical feature.
+    d(4, i) = mlpack::math::RandInt(0, 5);
+
+    // Mappings from categorical features to regression value.
+    std::map<int, double> f;
+    f[0] = 5.0;
+    f[1] = -5.0;
+
+    std::map<int, double> g;
+    g[0] = 2.0;
+    g[1] = 7.0;
+    g[2] = -3.0;
+    g[3] = 0.0;
+    g[4] = 4.0;
+
+    // Random noise in range [-0.5, 0.5).
+    const double noise = mlpack::math::Random() - 0.5;
```

Comment:
Nice, this looks good!  You might consider reducing the range of the noise a little bit, but if you're not seeing any failures then I think it's fine.

Reply:
I am getting pretty good RMSE (close to 0.5). So, I think this is just fine.

Reply:
Ah, ok, sounds good then. :+1: